minerl/Malmo/Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/ObservationFromRayImplementation.java

```diff
@@ -198,52 +198,46 @@ else if (mop.typeOfHit == RayTraceResult.Type.ENTITY)
         json.add("LineOfSight", jsonMop);
     }
 
-    static RayTraceResult findEntity(Vec3d eyePos, Vec3d lookVec, double depth, RayTraceResult mop, boolean includeTiles)
+    public static RayTraceResult findEntity(Vec3d eyePos, Vec3d lookVec, double depth, RayTraceResult mop, boolean includeTiles)
     {
         // Based on code in EntityRenderer.getMouseOver()
+
+        // Don't return entities through other blocks!
         if (mop != null)
             depth = mop.hitVec.distanceTo(eyePos);
-        Vec3d searchVec = eyePos.addVector(lookVec.xCoord * depth, lookVec.yCoord * depth, lookVec.zCoord * depth);
+
+        // Scale lookVec by the depth and don't assume it has been normalized
+        lookVec = lookVec.scale(depth / lookVec.lengthVector());
+
+        Vec3d searchVec = eyePos.add(lookVec);
         Entity pointedEntity = null;
 
         Vec3d hitVec = null;
         Entity viewer = Minecraft.getMinecraft().player;
-        List<?> list = Minecraft.getMinecraft().world.getEntitiesWithinAABBExcludingEntity(viewer, viewer.getEntityBoundingBox().addCoord(lookVec.xCoord * depth, lookVec.yCoord * depth, lookVec.zCoord * depth).expand(1.0, 1.0, 1.0));
+        List<Entity> list = Minecraft.getMinecraft().world.getEntitiesWithinAABBExcludingEntity(viewer, viewer.getEntityBoundingBox().addCoord(lookVec.xCoord, lookVec.yCoord, lookVec.zCoord).expand(1.0, 1.0, 1.0));
         double distance = depth;
 
-        for (int i = 0; i < list.size(); ++i)
-        {
-            Entity entity = (Entity)list.get(i);
-            if (entity.canBeCollidedWith() || includeTiles)
-            {
+        for (Entity entity : list) {
+            if (entity.canBeCollidedWith() || includeTiles) {
```

Comment:
I didn't look super closely at this, are further changes autoformatting?

Reply:
Yep!